### PR TITLE
Room

### PR DESCRIPTION
**Pull request by codefionn**
This migrates the existing database access solution to Android Room. The only thing that seems to be problematic is the "Restore from Backup" function. Though I did multiple bug fixes on top of that and works fine on my phone, I would keep an eye on it.

I also daily-drive this branch about a month now and it works fine.

### CODE REVIEW

**Comment by codefionn** [comment]
Idea: Restart the app after restoring from backup.

**Comment by gmathi** [comment]
Holy cow!! This needs a deep detailed review. So it will take some time for me. :)

**Review by gmathi — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 1]
```diff
@@ -97,7 +107,7 @@ dependencies {
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
     implementation 'androidx.appcompat:appcompat:1.2.0'
     implementation 'androidx.cardview:cardview:1.0.0'
-    implementation 'com.google.android.material:material:1.3.0'
+    implementation 'com.google.android.material:material:1.2.1'
```
[the remark this answers is not in the document]
Comment (codefionn) [reply app/build.gradle:110]: Merge error

**Review by codefionn — COMMENTED** [review]
Yeah, a merge error

**Comment by codefionn** [comment]
Fixed in #836f1d0

File: app/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (codefionn) [reply app/build.gradle]: Fixed by the way

**Comment by codefionn** [comment]
Discovered a bug: Sometimes chapters aren't marked as read

**Review by gmathi — APPROVED** [review]

**Comment by gmathi** [comment]
Found an issue. @codefionn In the "Downloads" Screen, Pausing a novel download will remove it from the list. If you relaunch the download screen, it does come back.

**Comment by codefionn** [comment]
> Found an issue. @codefionn In the "Downloads" Screen, Pausing a novel download will remove it from the list. If you relaunch the download screen, it does come back. 

will try to fix it

**Pull request merged**